### ISSUE TITLE
mavros version 1.5.0 doesn't publish /mavros/state topic

### ISSUE THREAD

**Issue by bigsuperZZZX** [mavlink/mavros#1518]
Current stable version 1.5.0 doesn't publish "/mavros/state" topic.
I installed mavros 1.5.0 following binary installation method and launched the px4.launch to connect to a pixhawk flight controller.
Various topics are publishing normally except the "/mavros/state" topic
![image](https://user-images.githubusercontent.com/26641182/103100736-011d7300-464f-11eb-8737-241b8c6ec766.png).

Following tests are done:
1. Three different FCUs: Pixracer, CUAV V5+, Holybro pix32 v5. None of which make the "/mavros/state" published.
2. An older version of mavros is tested: a source installed mavros 1.0.0. That topic published normaly.
3. Two different ubuntu version 16.04 and 18.04 are tested, On none of them that topic published.

### MAVROS version and platform
Mavros: 1.5.0
ROS: Kinetic and Melodic
Ubuntu: 16.04 and 18.04

### Autopilot type and version
PX4
Version: 1.11.2

### Node logs
Everything seems normal.
```
process[mavros-2]: started with pid [65618]
[ INFO] [1608831037.858531027]: FCU URL: /dev/ttyACM0:921600
[ INFO] [1608831037.860076302]: serial0: device: /dev/ttyACM0 @ 921600 bps
[ INFO] [1608831037.862222815]: GCS bridge disabled
[ INFO] [1608831037.867997562]: Plugin 3dr_radio loaded
[ INFO] [1608831037.869556745]: Plugin 3dr_radio initialized
[ INFO] [1608831037.869631167]: Plugin actuator_control loaded
[ INFO] [1608831037.871592256]: Plugin actuator_control initialized
[ INFO] [1608831037.873414478]: Plugin adsb loaded
[ INFO] [1608831037.875581494]: Plugin adsb initialized
[ INFO] [1608831037.875661597]: Plugin altitude loaded
[ INFO] [1608831037.876261475]: Plugin altitude initialized
[ INFO] [1608831037.876336522]: Plugin cam_imu_sync loaded
[ INFO] [1608831037.876785436]: Plugin cam_imu_sync initialized
[ INFO] [1608831037.876860874]: Plugin command loaded
[ INFO] [1608831037.883336253]: Plugin command initialized
[ INFO] [1608831037.883428523]: Plugin companion_process_status loaded
[ INFO] [1608831037.885258836]: Plugin companion_process_status initialized
[ INFO] [1608831037.885327278]: Plugin debug_value loaded
[ INFO] [1608831037.888549823]: Plugin debug_value initialized
[ INFO] [1608831037.888614045]: Plugin distance_sensor blacklisted
[ INFO] [1608831037.888684083]: Plugin esc_status loaded
[ INFO] [1608831037.889434952]: Plugin esc_status initialized
[ INFO] [1608831037.889522007]: Plugin fake_gps loaded
[ INFO] [1608831037.899830490]: Plugin fake_gps initialized
[ INFO] [1608831037.899933350]: Plugin ftp loaded
[ INFO] [1608831037.903550333]: Plugin ftp initialized
[ INFO] [1608831037.903649064]: Plugin global_position loaded
[ INFO] [1608831037.912167910]: Plugin global_position initialized
[ INFO] [1608831037.912258160]: Plugin gps_rtk loaded
[ INFO] [1608831037.913790923]: Plugin gps_rtk initialized
[ INFO] [1608831037.913865842]: Plugin gps_status loaded
[ INFO] [1608831037.914997976]: Plugin gps_status initialized
[ INFO] [1608831037.915082159]: Plugin hil loaded
[ INFO] [1608831037.923142321]: Plugin hil initialized
[ INFO] [1608831037.923234188]: Plugin home_position loaded
[ INFO] [1608831037.925944138]: Plugin home_position initialized
[ INFO] [1608831037.926035251]: Plugin imu loaded
[ INFO] [1608831037.930134223]: Plugin imu initialized
[ INFO] [1608831037.930226081]: Plugin landing_target loaded
[ INFO] [1608831037.938383393]: Plugin landing_target initialized
[ INFO] [1608831037.938489432]: Plugin local_position loaded
[ INFO] [1608831037.942574825]: Plugin local_position initialized
[ INFO] [1608831037.942674874]: Plugin log_transfer loaded
[ INFO] [1608831037.944080686]: Plugin log_transfer initialized
[ INFO] [1608831037.944165446]: Plugin manual_control loaded
[ INFO] [1608831037.945715494]: Plugin manual_control initialized
[ INFO] [1608831037.945872853]: Plugin mocap_pose_estimate loaded
[ INFO] [1608831037.948084169]: Plugin mocap_pose_estimate initialized
[ INFO] [1608831037.948145420]: Plugin mount_control loaded
[ INFO] [1608831037.950555134]: Plugin mount_control initialized
[ INFO] [1608831037.950663451]: Plugin obstacle_distance loaded
[ INFO] [1608831037.952165112]: Plugin obstacle_distance initialized
[ INFO] [1608831037.952251812]: Plugin odom loaded
[ INFO] [1608831037.954864255]: Plugin odom initialized
[ INFO] [1608831037.954955753]: Plugin onboard_computer_status loaded
[ INFO] [1608831037.956197432]: Plugin onboard_computer_status initialized
[ INFO] [1608831037.956279945]: Plugin param loaded
[ INFO] [1608831037.958020451]: Plugin param initialized
[ INFO] [1608831037.958108555]: Plugin play_tune loaded
[ INFO] [1608831037.959844383]: Plugin play_tune initialized
[ INFO] [1608831037.959926972]: Plugin px4flow loaded
[ INFO] [1608831037.963770436]: Plugin px4flow initialized
[ INFO] [1608831037.963829602]: Plugin rangefinder blacklisted
[ INFO] [1608831037.963903555]: Plugin rc_io loaded
[ INFO] [1608831037.965816402]: Plugin rc_io initialized
[ INFO] [1608831037.965858780]: Plugin safety_area blacklisted
[ INFO] [1608831037.965913816]: Plugin setpoint_accel loaded
[ INFO] [1608831037.967965165]: Plugin setpoint_accel initialized
[ INFO] [1608831037.968058282]: Plugin setpoint_attitude loaded
[ INFO] [1608831037.973097183]: Plugin setpoint_attitude initialized
[ INFO] [1608831037.973193957]: Plugin setpoint_position loaded
[ INFO] [1608831037.982201440]: Plugin setpoint_position initialized
[ INFO] [1608831037.982316304]: Plugin setpoint_raw loaded
[ INFO] [1608831037.987458629]: Plugin setpoint_raw initialized
[ INFO] [1608831037.987580120]: Plugin setpoint_trajectory loaded
[ INFO] [1608831037.990518895]: Plugin setpoint_trajectory initialized
[ INFO] [1608831037.990587031]: Plugin setpoint_velocity loaded
[ INFO] [1608831037.993554086]: Plugin setpoint_velocity initialized
[ INFO] [1608831037.993651904]: Plugin sys_status loaded
[ INFO] [1608831037.999648903]: Plugin sys_status initialized
[ INFO] [1608831037.999748410]: Plugin sys_time loaded
[ INFO] [1608831038.002723762]: TM: Timesync mode: MAVLINK
[ INFO] [1608831038.003384452]: Plugin sys_time initialized
[ INFO] [1608831038.003488383]: Plugin trajectory loaded
[ INFO] [1608831038.006882970]: Plugin trajectory initialized
[ INFO] [1608831038.006984810]: Plugin vfr_hud loaded
[ INFO] [1608831038.007309795]: Plugin vfr_hud initialized
[ INFO] [1608831038.007366883]: Plugin vibration blacklisted
[ INFO] [1608831038.007436402]: Plugin vision_pose_estimate loaded
[ INFO] [1608831038.011400287]: Plugin vision_pose_estimate initialized
[ INFO] [1608831038.011503538]: Plugin vision_speed_estimate loaded
[ INFO] [1608831038.014556392]: Plugin vision_speed_estimate initialized
[ INFO] [1608831038.014694934]: Plugin waypoint loaded
[ INFO] [1608831038.017284214]: Plugin waypoint initialized
[ INFO] [1608831038.017345293]: Plugin wheel_odometry blacklisted
[ INFO] [1608831038.017423352]: Plugin wind_estimation loaded
[ INFO] [160881038.017827866]: Plugin wind_estimation initialized
[ INFO] [1608831038.017887905]: Autostarting mavlink via USB on PX4
[ INFO] [1608831038.017962505]: Built-in SIMD instructions: SSE, SSE2
[ INFO] [1608831038.018011477]: Built-in MAVLink package version: 2020.12.12
[ INFO] [1608831038.018042905]: Known MAVLink dialects: common ardupilotmega ASLUAV all autoquad icarous matrixpilot paparazzi standard uAvionix ualberta
[ INFO] [1608831038.018093324]: MAVROS started. MY ID 1.240, TARGET ID 1.1
[ INFO] [1608831038.032070423]: IMU: Attitude quaternion IMU detected!
[ INFO] [1608831038.033615670]: IMU: High resolution IMU detected!

```

### Diagnostics
```
header: 
  seq: 8
  stamp: 
    secs: 1608830959
    nsecs: 430560725
  frame_id: ''
status: 
  - 
    level: 1
    name: "mavros: FCU connection"
    message: "not connected"
    hardware_id: "/dev/ttyACM0:921600"
    values: 
      - 
        key: "Received packets:"
        value: "1450"
      - 
        key: "Dropped packets:"
        value: "0"
      - 
        key: "Buffer overruns:"
        value: "0"
      - 
        key: "Parse errors:"
        value: "0"
      - 
        key: "Rx sequence number:"
        value: "239"
      - 
        key: "Tx sequence number:"
        value: "0"
      - 
        key: "Rx total bytes:"
        value: "74150"
      - 
        key: "Tx total bytes:"
        value: "1857"
      - 
        key: "Rx speed:"
        value: "32684.000000"
      - 
        key: "Tx speed:"
        value: "850.000000"
  - 
    level: 2
    name: "mavros: GPS"
    message: "No satellites"
    hardware_id: "/dev/ttyACM0:921600"
    values: 
      - 
        key: "Satellites visible"
        value: "0"
      - 
        key: "Fix type"
        value: "0"
      - 
        key: "EPH (m)"
        value: "Unknown"
      - 
        key: "EPV (m)"
        value: "Unknown"
  - 
    level: 2
    name: "mavros: Heartbeat"
    message: "No events recorded."
    hardware_id: "/dev/ttyACM0:921600"
    values: 
      - 
        key: "Heartbeats since startup"
        value: "0"
      - 
        key: "Frequency (Hz)"
        value: "0.000000"
      - 
        key: "Vehicle type"
        value: "Generic micro air vehicle"
      - 
        key: "Autopilot type"
        value: "Generic autopilot"
      - 
        key: "Mode"
        value: ''
      - 
        key: "System status"
        value: "Uninit"
  - 
    level: 2
    name: "mavros: System"
    message: "Sensor health"
    hardware_id: "/dev/ttyACM0:921600"
    values: 
      - 
        key: "Sensor present"
        value: "0x12000008"
      - 
        key: "Sensor enabled"
        value: "0x12010000"
      - 
        key: "Sensor health"
        value: "0x0006000B"
      - 
        key: "rc receiver"
        value: "Fail"
      - 
        key: "Battery"
        value: "Fail"
      - 
        key: "CPU Load (%)"
        value: "43.4"
      - 
        key: "Drop rate (%)"
        value: "0.0"
      - 
        key: "Errors comm"
        value: "0"
      - 
        key: "Errors count #1"
        value: "0"
      - 
        key: "Errors count #2"
        value: "0"
      - 
        key: "Errors count #3"
        value: "0"
      - 
        key: "Errors count #4"
        value: "0"
  - 
    level: 0
    name: "mavros: Battery"
    message: "Normal"
    hardware_id: "/dev/ttyACM0:921600"
    values: 
      - 
        key: "Voltage"
        value: "65.54"
      - 
        key: "Current"
        value: "-0.0"
      - 
        key: "Remaining"
        value: "-1.0"
  - 
    level: 0
    name: "mavros: Time Sync"
    message: "Normal"
    hardware_id: "/dev/ttyACM0:921600"
    values: 
      - 
        key: "Timesyncs since startup"
        value: "35"
      - 
        key: "Frequency (Hz)"
        value: "9.946536"
      - 
        key: "Last RTT (ms)"
        value: "2.111450"
      - 
        key: "Mean RTT (ms)"
        value: "1.922639"
      - 
        key: "Last remote time (s)"
        value: "1255.658136000"
      - 
        key: "Estimated time offset (s)"
        value: "1608829703.759995222"
---

```

### Check ID
```
ERROR. I got 0 addresses, but not your target 1:1

---
Received 0 messages, from 0 addresses
sys:comp   list of messages
```

Thank you for your reply!

**Comment by vooon** [comment]
That looks like you didn't get HEARTBEAT's. I'm not quite sure where to look because there are almost no changes in libmavconn since 1.0.
Although one big change comes from mavlink itself with moves of some messages from common.

Could you try to get latest mavros and mavlink 2020.12.12 and try it with freshest PX4?
Also please look into `/mavlink/from` topic, it should have some stream of messages. From `check id` it looks like completely silent.

**Comment by bigsuperZZZX** [comment]
Following the given hints, I think I've found the solution, which seems really wired but helpful.

After various trials and comparisons, I found that the problem comes from the SD card in FCU. All tested FCUs have no SD card installed since last firmware burning. As soon as I insert a SD card and reboot the FCU, that problem disappeared and the required topic published normally. After that, I removed the SD card again and rebooted the FCU, the mavros can still work well. 
So I think the solution to solve this problem is just inserting a SD card especially after the first firmware burning or parameter changing.

Thank you for your hints!
